### PR TITLE
Add simple logging to file, add Elk stack, fix very minor boring issue in the login_controller

### PR DESCRIPTION
**Pull request by JeppeV**
This PR does a few things:
1. It adds Rack middleware logging + event/error logging into the app and api.
2. It adds the ELK stack which allows easy overview and query of logs (3 docker containers to run on the server)
3. It fixes some minor bugs

This closes: #139, #127

### CODE REVIEW

**Comment by JeppeV** [comment]
This also includes the content of PR #141

File: app/controllers/login_controller.rb [hunk 1]
```diff
@@ -13,10 +14,9 @@ def attempt_login_user
     error = nil
     if user.nil?
       error = 'Invalid username'
-    elsif !user.password == @request.params['password']
+    elsif BCrypt::Password.new(user.password) != @request.params['password']
       error = 'Invalid password'
```
Comment (JeppeV) [inline app/controllers/login_controller.rb]: This also fixes the fact that our original non-functional password check didn't actually check the hash, but the plain password instead. The problem only came to light after we started checking for password equality after the other bug was fixed.

File: api/Rakefile [hunk 2]
```diff
@@ -52,3 +53,13 @@ namespace :db do
     sh "rm #{databases['production']['database']}"
   end
 end
+
+namespace :log do
+  log_config = YAML.load_file('config/log.yml')
+  task :create do
+    puts 'Creating API logs …'
+    sh 'mkdir /var/www/log' unless File.directory?('/var/www/log')
```
Comment (janschill) [inline api/Rakefile]: Have a look at this:

Folder creation (recursively)

```ruby
require 'fileutils'

dirname = File.dirname(some_path)
unless File.directory?(dirname)
  FileUtils.mkdir_p(dirname)
end
```

File creation for writing

```ruby
out_file = File.new("out.txt", "w")
```

File: api/config.ru [hunk 3]
```diff
@@ -9,6 +9,7 @@ require 'logger'
 logger = Logger.new(log_config['access']['filepath'])
 use Rack::CommonLogger, logger
 
+# require 'prometheus/middleware/collector'
```
Comment (janschill) [inline api/config.ru]: This is now twice in the file.

File: app/Rakefile [hunk 4]
```diff
@@ -98,3 +99,13 @@ namespace :env do
     end
   end
 end
+
+namespace :log do
+  log_config = YAML.load_file('config/log.yml')
+  task :create do
+    puts 'Creating APP logs …'
+    sh 'mkdir /var/www/log' unless File.directory?('/var/www/log')
```
Comment (janschill) [inline app/Rakefile]: See comment before regarding creation of files/folders in Ruby.

**Pull request merged**